app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -1,17 +1,104 @@
-// package ch.epfl.favo.common;
-//
-// import com.google.firebase.firestore.FirebaseFirestore;
-//
-// public final class DatabaseWrapper {
-//
-//    private static FirebaseFirestore mFirestore;
-//
-//    public DatabaseWrapper() {
-//        FirebaseFirestore.setLoggingEnabled(true);
-//        mFirestore = FirebaseFirestore.getInstance();
-//    }
-//
-//    public static FirebaseFirestore getDatabaseInstance() {
-//        return mFirestore;
-//    }
-// }
+package ch.epfl.favo.common;
+
+import android.util.Log;
+
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreSettings;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class DatabaseWrapper<T extends DatabaseDocument> {
+
+    private static final String TAG = "DatabaseWrapper";
+
+    private static FirebaseFirestore firestore;
+
+    private Class<T> cls;
```

Comment:
You could delegate the type parameter to the function directly (`public <T> T getDocument(...)`). Imo, you could have the class not store any state and use a static class instead, by passingyour current properties as function parameters. But this is arguable.